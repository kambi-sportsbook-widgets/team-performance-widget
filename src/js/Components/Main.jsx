import React from 'react';
import Header from './Header';

const Main = ({ title, children, defaultExpanded, expandHandler }) => {
   const onChangeHandler = (event) => {
      if (expandHandler) {
         expandHandler(event.target.checked);
      }
   };

   return (
      <main className="kw-main">
         <Header title={title} />
         <input className="kw-accordion" type="checkbox" defaultChecked={defaultExpanded} onChange={onChangeHandler} />
         {children}
      </main>
   );
};

Main.propTypes = {
   /**
    * Widget's title
    */
   title: React.PropTypes.string.isRequired,

   /**
    * Inner components
    */
   children: React.PropTypes.arrayOf(React.PropTypes.element).isRequired,

   /**
    * Should details be expanded on the beginning
    */
   defaultExpanded: React.PropTypes.bool,

   /**
    * Called on expand click, gets one 'currentState' argument
    */
   expandHandler: React.PropTypes.func
};

Main.defaultProps = {
   defaultExpanded: false
};

export default Main;
