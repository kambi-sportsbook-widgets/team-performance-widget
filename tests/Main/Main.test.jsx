/* eslint-env jest */
import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import Main from '../../src/js/Components/Main';

let renderer;

describe('Main DOM rendering', () => {
   beforeEach(() => {
      renderer = ReactTestUtils.createRenderer();
   });
   it('renders correctly', () => {
      expect(renderer.render(
         <Main title='TheTitle'>
            <span>content</span>
         </Main>
      )).toMatchSnapshot();
   });
});
