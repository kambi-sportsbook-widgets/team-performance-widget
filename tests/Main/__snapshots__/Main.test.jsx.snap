// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main DOM rendering renders correctly 1`] = `
<main
  className="kw-main"
>
  <Header
    title="TheTitle"
  />
  <span>
    content
  </span>
</main>
`;
